src/Workspaces/Core/Portable/Rename/RenameSymbolAnnotation.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    /// <summary>
+    /// Provides a way to produce and consume annotations for rename
+    /// that contain the previous symbol serialized as a <see cref="SymbolKey" />.
+    /// This annotations is used by <see cref="Workspace.TryApplyChanges(Solution)" /> 
+    /// in some cases to notify the workspace host of refactorings.
+    /// </summary>
+    /// <remarks>
+    /// This annotation is applied to the declaring syntax of a symbol that has been renamed. 
+    /// When Workspace.TryApplyChanges happens in Visual Studio, we raise rename events for that symbol.
+    /// </remarks>
+    internal static class RenameSymbolAnnotation
+    {
+        public const string RenameSymbolKind = nameof(RenameSymbolAnnotation);
+
+        /// <summary>
+        /// Returns true if there are no existing rename annotations
+        /// and the symbol is a candidate for annotation. Out parameter
+        /// is the node with the annotation on it
+        /// </summary>
+        public static bool TryAnnotateNode(SyntaxNode syntaxNode, ISymbol? symbol, [NotNullWhen(returnValue: true)] out SyntaxNode? annotatedNode)
+        {
+            annotatedNode = null;
+            if (!ShouldAnnotateSymbol(symbol))
+            {
+                return false;
+            }
+
+            // If the node is already annotated, assume the original annotation is 
+            // more correct for representing the original symbol
+            if (syntaxNode.GetAnnotations(RenameSymbolKind).Any())
```

Comment:
when does this happen?  i'm confused about the data flow here.  when are we trying to annotate an already annotated symbol?